frontend/src/components/SocialSidebar/MiniMessageBox.tsx

```diff
@@ -0,0 +1,98 @@
+import { Button, Input, useToast, VStack } from '@chakra-ui/react';
+import React, { useEffect, useState } from 'react'; // gets rid of eslint error
+import { useSelectedFriends } from '../../classes/TownController';
+import useTownController from '../../hooks/useTownController';
+import { MiniMessage } from '../../types/CoveyTownSocket';
+
+/**
+ * Creates a Chakra Button and Input text box that allows a given player to send a MiniMessage
+ * to their currently selected friends.
+ * @returns {JSX.Element} a Chakra VStack containing an input box and button that sends the
+ *                        message input to the selected players as a Toast
+ */
+export default function MiniMessageBox(): JSX.Element {
+  const townController = useTownController();
+  const selectedFriends = useSelectedFriends();
+  const [miniMessageBody, setMiniMessageBody] = useState<string>('');
+  const [isTextareaFocused, setIsTextareaFocused] = useState(false);
+  const toast = useToast();
+
+  const attemptSendMessage = () => {
+    if (miniMessageBody !== '') {
+      if (selectedFriends.length > 0) {
+        const miniMessageToSend: MiniMessage = {
+          sender: townController.ourPlayer.id,
+          recipients: selectedFriends.map(friend => friend.id),
+          body: miniMessageBody,
+        };
+        townController.clickedSendMiniMessage(miniMessageToSend);
+        toast({
+          title: 'MiniMessage sent!',
+          description: '',
+          status: 'success',
+          duration: 3000,
+          isClosable: true,
+          position: 'top',
+        });
+      } else {
+        toast({
+          title: 'No recipients selected',
+          description: 'Must select at least one friend to send a MiniMessage to.',
+          status: 'error',
+          duration: 9000,
+          isClosable: true,
+          position: 'top',
+        });
+      }
+    } else {
+      toast({
+        title: 'Cannot send blank message',
+        description: 'Enter a message to send a MiniMessage.',
+        status: 'error',
+        duration: 9000,
+        isClosable: true,
+        position: 'top',
+      });
+    }
+  };
+
+  // Allows player to move again once they have clicked out of the input box
+  useEffect(() => {
+    if (isTextareaFocused) {
+      townController.pause();
+    } else {
+      townController.unPause();
+    }
+  }, [isTextareaFocused, townController]);
+
+  // Allows "enter" key to send the message
+  const handleReturnKeyPress = (event: React.KeyboardEvent) => {
+    if (isTextareaFocused && event.key === 'Enter') {
+      event.preventDefault();
+      attemptSendMessage();
+    }
+  };
+
+  return (
+    <VStack align='left' mt='2'>
+      <Input
+        borderColor='grey.900'
+        rounded='md'
+        size='sm'
+        placeholder='Write a MiniMessage...'
```

Comment:
```suggestion
        placeholder='Write a Mini Message...'
```

Reply:
Since I'm thinking of "MiniMessage" as its own thing (like a "Tweet"), and not just a message that is mini-sized, I think it makes sense to keep it as the combined name "MiniMessage".

Reply:
oh yeah thats fun! sounds good